### PR TITLE
Popup

### PR DESCRIPTION
**Pull request by oskovbasiuk**

### CODE REVIEW

File: client/src/modules/MapHolder/components/PopupHolder.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+              {Object.keys(defInfo).map(key => {
+                let field = defInfo[key];
+                if (key === 'actual_date') {
+                  const dateDelim = field.indexOf('T');
+                  const timeDelim = field.indexOf('Z');
+                  field =
+                    field.slice(0, dateDelim) +
+                    ' ' +
+                    field.slice(dateDelim + 1, timeDelim);
+                }
```
Comment (yuriiholskiy) [inline client/src/modules/MapHolder/components/PopupHolder.js]: Maybe use for date format Intl.DateTimeFormat. 
Link to mdn - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat

File: client/src/modules/MapHolder/components/PopupContent/DefibrillatorPopupContent.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  }
+});
+const DefibrillatorPopupContent = ({ id }) => {
+  const classes = useStyle();
+
+  const [currDef, setCurrDef] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const res = await http.get(
```
Comment (OrestSat) [inline client/src/modules/MapHolder/components/PopupContent/DefibrillatorPopupContent.js]: Move to api layer

File: client/src/modules/MapHolder/components/PopupHolder.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import DefContent from './PopupContent/DefibrillatorPopupContent';
+
+const PopupHolder = ({ popupData }) => {
+  return (
+    <>
+      {popupData ? (
+        <Popup
+          coordinates={popupData.coordinates}
+          offset={{
+            // constant
```
Comment (OrestSat) [inline client/src/modules/MapHolder/components/PopupHolder.js]: Move to consts

File: client/src/modules/MapHolder/index.js [hunk 4]
```diff
@@ -1,120 +1,96 @@
...
       zoom={[zoom]}
-      onStyleLoad={rawMap => {
-        if (rawMap) {
-          loadMap(rawMap);
-        }
+      onStyleLoad={loadMap}
+      onZoomEnd={onZoomEnded}
+      onClick={onClick}
+      onZoomStart={() => {
+        onZoomStarted();
```
Comment (OrestSat) [inline client/src/modules/MapHolder/index.js]: Not needed to wrap

Comment (oskovbasiuk) [reply client/src/modules/MapHolder/index.js]: done

File: client/src/modules/MapHolder/components/PopupHolder.js (hunk 3 above)
Comment (oskovbasiuk) [reply client/src/modules/MapHolder/components/PopupHolder.js]: done

File: client/src/modules/MapHolder/components/PopupContent/DefibrillatorPopupContent.js (hunk 2 above)
Comment (oskovbasiuk) [reply client/src/modules/MapHolder/components/PopupContent/DefibrillatorPopupContent.js]: done

File: client/src/modules/MapHolder/components/PopupContent/DefibrillatorPopupContent.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  const [currDef, setCurrDef] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setCurrDef(null);
+      const res = await fetchSingleDefById(id);
+      const { defibrillator } = res.data;
+      setCurrDef(defibrillator);
+    };
+    fetchData();
```
Comment (JaroSlavDEV) [inline client/src/modules/MapHolder/components/PopupContent/DefibrillatorPopupContent.js]: Add canceltoken on unmount

File: routes/defRoute.js [hunk 6]
```diff
@@ -66,10 +66,25 @@ router.put(
...
+router.get('/:id', async (req, res) => {
+  try {
+    const { id } = req.params;
+    const defibrillator = await Defibrillator.findById(id);
+    return res.status(200).send({
+      error: false,
+      defibrillator
+    });
+  } catch (e) {
+    errorHandler(res, e);
```
Comment (JaroSlavDEV) [inline routes/defRoute.js]: In the next pull request i will change this function.
So change it to - resServerError(res, e);

**Pull request merged**